File: python/folderprogram/Gadget.py
# Gadget.py

from ProdukElektronik import ProdukElektronik

class Gadget(ProdukElektronik):
    def __init__(self, kode, nama, harga, deskripsi, jenis, garansi, status, stok):
        # FIXED: Meneruskan 'kode' ke parent class
        super().__init__(kode, nama, harga, deskripsi)
        self.jenis = jenis
        self.garansi = garansi
        self.status = status
        self.stok = stok
        
    # ADDED: Metode display_info untuk kelas Gadget
    def display_info(self):
        super().display_info()
        print(f"Jenis: {self.jenis}")
        print(f"Garansi: {self.garansi}")
        print(f"Status: {self.status}")
        print(f"Stok: {self.stok}")

    # FIXED: Getter dan Setter dipindahkan ke dalam kelas
    def get_jenis(self):
        return self.jenis

    def set_jenis(self, jenis):
        self.jenis = jenis

    def get_garansi(self):
        return self.garansi

    def set_garansi(self, garansi):
        self.garansi = garansi

    def get_status(self):
        return self.status

    def set_status(self, status):
        self.status = status

    def get_stok(self):
        return self.stok

    def set_stok(self, stok):
        self.stok = stok

File: php/folderprogram/Gadget.php
<?php
require_once "ProdukElektronik.php";

class Gadget extends ProdukElektronik {
    private $jenis;
    private $garansi;
    private $status;
    private $stok;
    private $foto;

    public function __construct($kode = "", $nama = "", $harga = 0, $deskripsi = "",
                                $jenis = "", $garansi = "", $status = "", $stok = 0,$foto = ""  ) {
        parent::__construct($kode,$nama, $harga, $deskripsi);
        $this->jenis = $jenis;
        $this->garansi = $garansi;
        $this->status = $status;
        $this->stok = $stok;
        $this->foto = $foto;
    }

    // Getter & Setter
    public function getJenis() { return $this->jenis; }
    public function setJenis($jenis) { $this->jenis = $jenis; }

    public function getGaransi() { return $this->garansi; }
    public function setGaransi($garansi) { $this->garansi = $garansi; }

    public function getStatus() { return $this->status; }
    public function setStatus($status) { $this->status = $status; }

    public function getStok() { return $this->stok; }
    public function setStok($stok) { $this->stok = $stok; }

     public function getFoto() { return $this->foto; } // Tambahkan getter untuk $foto
    public function setFoto($foto) { $this->foto = $foto; } // Tambahkan setter untuk $foto


    public function displayInfo() {
        parent::displayInfo();
        echo "Jenis: {$this->jenis}<br>";
        echo "Garansi: {$this->garansi}<br>";
        echo "Status: {$this->status}<br>";
        echo "Stok: {$this->stok}<br>";
        echo "Foto: {$this->foto}<br>";
    }
}
?>
